crates/zk-por-core/src/parser.rs

```diff
@@ -1,11 +1,173 @@
 use std::{collections::BTreeMap, fs::File, io::BufReader};
 
+use super::account::Account;
 use crate::types::F;
 use plonky2_field::types::Field;
 use serde_json::Value;
-use tracing::error;
+use std::{
+    fs,
+    ops::Div,
+    path::{Path, PathBuf},
+};
+use tracing::{debug, error, info, warn};
 
-use super::account::Account;
+#[derive(Debug)]
+pub struct FilesCfg {
+    pub dir: PathBuf,
+    pub batch_size: usize,
+    pub num_of_tokens: usize,
+}
+
+#[derive(Debug)]
+pub struct FilesParser {
+    pub cfg: FilesCfg,
+    pub total_num_of_batches: usize,
+    pub total_num_of_users: usize,
+    num_of_docs: usize,
+    num_of_batches_per_doc: usize,
+    file_idx: usize,
+    offset: usize, // an offset to the buffered_accounts
+    buffered_accounts: Vec<Account>,
+    last_doc_accounts: Vec<Account>, // we keep this avoiding double loading of last doc
+    docs: Vec<PathBuf>,
+}
+
+impl FilesParser {
+    pub fn new(cfg: FilesCfg) -> Self {
+        let mut parser = Self {
+            buffered_accounts: vec![],
+            last_doc_accounts: vec![],
+            file_idx: 0,
+            offset: 0,
+            num_of_docs: 0,
+            cfg,
+            num_of_batches_per_doc: 0,
+            total_num_of_users: 0,
+            total_num_of_batches: 0,
+            docs: vec![],
+        };
+        let user_data_path = std::path::Path::new(&parser.cfg.dir);
+        if !user_data_path.exists() {
+            panic!("dir: {:?} does not exist", user_data_path);
+        }
+
+        let json_files = list_json_files(&user_data_path);
+
+        match json_files {
+            Ok(docs) => {
+                info!("files: {:?}", docs);
+                let doc_len = docs.len();
+                parser.docs = docs;
+                parser.num_of_docs = doc_len;
+                if doc_len < 1 {
+                    warn!("no json files under the folder: {:?}", user_data_path);
+                    std::process::exit(0);
+                } else {
```

Comment:
no need to wrap the code in ``else`` block, as the program will exit in the above ``if`` block.